File: 2018-08-09/W2D4/W2D4/SecondViewController.m
//
//  SecondViewController.m
//  W2D4
//
//  Created by Jason Liang on 8/9/18.
//  Copyright © 2018 jasonliang. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()
@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation SecondViewController


- (IBAction)handleTapGesture:(id)sender {
  NSLog(@"Label tapped");
}

- (IBAction)handlePanGesture:(UIPanGestureRecognizer *)sender {
  if (sender.state == UIGestureRecognizerStateChanged) {
    self.label.center = [sender locationInView:self.view];
  }
}


@end

File: 2018-03-01/W7D4/W7D4/SecondViewController.swift
//
//  SecondViewController.swift
//  W7D4
//
//  Created by Jason Liang on 3/1/18.
//  Copyright © 2018 jasonliang. All rights reserved.
//

import UIKit

class SecondViewController: UIViewController {

    @IBOutlet weak var stackView: UIStackView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    @IBAction func addItem(_ sender: Any) {
        let frame = CGRect(x: 0, y: 0, width: self.view.bounds.width, height: 150)
        let item = ItemView(frame: frame)
        item.translatesAutoresizingMaskIntoConstraints = false
        item.widthAnchor.constraint(equalToConstant: self.view.bounds.width).isActive = true
        item.heightAnchor.constraint(equalToConstant: 150).isActive = true
        stackView.addArrangedSubview(item)
        item.delegate = self
    }
}

extension SecondViewController : ItemViewDelegate {
    func removeItem(itemView: UIView) {
//        itemView.removeFromSuperview()
        itemView.isHidden = true
//        stackView.removeArrangedSubview(itemView)
    }
}

